ModulePresentationsForCAP/gap/ModulePresentationsForCAP.gi

```diff
@@ -2146,3 +2172,336 @@ end );
+          fi;
+
+          u := UnderlyingMatrix( U );
+
+          x := homalg_ring!.random_matrix_func( n, NrCols( u ) );
+
+          syz := SyzygiesOfRows( UnionOfRows( x, u ) );
+
+          if NrRows( syz ) > 0 then
+
+            e := homalg_ring!.random_element_func( );
+
+            e := Random( [ e, e, e, One( homalg_ring ) ] );
+
+            syz := e * CertainRows( syz, [ 1 .. Random( [ 1 .. NrRows( syz ) ] ) ] );
+
+          fi;
+
+          m := CertainColumns( syz, [ 1 .. NrRows( x ) ] );
+
+          M := AsLeftPresentation( m );
+
+          return PresentationMorphism( M, x, U );
+
+    end );
+
+end );
+
+##
+InstallGlobalFunction( ADD_RANDOM_MORPHISM_WITH_FIXED_SOURCE_RIGHT,
+    function( category )
+      local homalg_ring;
+      
+      homalg_ring := category!.ring_for_representation_category;
+
+      AddRandomMorphismWithFixedSource( category,
+        function( U, n )
+          local u, x, syz, e, m, M, y;
+
+          if n < 0 then
+            
+            return fail;
+
+          fi;
+
+          u := UnderlyingMatrix( U );
+
+          x := homalg_ring!.random_matrix_func( n, NrCols( u ) );
+
+          syz := SyzygiesOfRows( UnionOfRows( x, u ) );
+
+          if NrRows( syz ) > 0 then
+
+            e := homalg_ring!.random_element_func( );
+
+            e := Random( [ e, e, e, One( homalg_ring ) ] );
+
+            syz := e * CertainRows( syz, [ 1 .. Random( [ 1 .. NrRows( syz ) ] ) ] );
+
+          fi;
+
+          m := CertainColumns( syz, [ 1 .. NrRows( x ) ] );
+
+          M := AsRightPresentation( m );
+
+          y := CertainColumns( syz, [ NrRows( x ) + 1 .. NrCols( syz ) ] );
+
+          # Now: m*x + y*u = 0
+          return PresentationMorphism( U, y, M );
+
+    end );
+
+end );
+
+##
+InstallGlobalFunction( ADD_RANDOM_MORPHISM_WITH_FIXED_RANGE_RIGHT,
+    function( category )
+      local homalg_ring;
+
+      homalg_ring := category!.ring_for_representation_category;
+
+      AddRandomMorphismWithFixedRange( category,
+        function( M, n )
+          local m, y, syz, e, u, U;
+          
+          if n < 0 then
+            
+            return fail;
+
+          fi;
+
+          if n = 0 then
+
+            return UniversalMorphismFromZeroObject( M );
+
+          fi;
+
+          m := UnderlyingMatrix( M );
+
+          y := homalg_ring!.random_matrix_func( NrRows( m ), n );
+
+          syz := SyzygiesOfColumns( UnionOfColumns( m, y ) );
+
+          if NrCols( syz ) > 0 then
+
+            e := homalg_ring!.random_element_func( );
+
+            e := Random( [ e, e, e, One( homalg_ring ) ] );
+
+            syz := CertainColumns( syz, [ 1 .. Random( [ 1 .. NrCols( syz ) ]  ) ] ) * e;
+
+          fi;
+
+          u := CertainRows( syz, [ NrCols( m ) + 1 .. NrRows( syz ) ] );
+
+          U := AsRightPresentation( u );
+
+          return PresentationMorphism( U, y, M );
+     
+      end );
+
+end );
+
+##
+InstallGlobalFunction( TRY_TO_ENHANCE_HOMALG_RING_WITH_RANDOM_FUNCTIONS,
```

Comment:
@mohamed-barakat and @kamalsaleh :
it would be great if these random functions for HomalgRings could be transfered to `RingsForHomalg`. In that way, other packages, like `FreydCategoriesForCAP,` could also profit. 

@kamalsaleh:
Another example how other packages can profit: the method `RandomMorphismWithFixedSource` in `LinearAlgebraForCAP` could simply use underlying random functions of the given field. At the moment, it simply produces integers interpreted in the given field in a hard coded range between `n-10` and `n+10`.

Reply:
`GradedRingForHomalg` package is the natural place for these methods. I talked with @mohamed-barakat about adding them or similar methods to homalg and he told me we can work on that when he comes back.

Reply:
The range in random methods of linear algebra package is `[-n-10, n+10]` for `n>0`.